Index: src/main/resources/config/spring/rc/applicationContext-ws-service.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/config/spring/rc/applicationContext-ws-service.xml	(revision 8440)
+++ src/main/resources/config/spring/rc/applicationContext-ws-service.xml	(revision )
@@ -15,6 +15,7 @@
 	<bean id="creditWS" class="com.creditease.webservice.impl.CreditWS" />
 	<bean id="comprehensiveCreditWS" class="com.creditease.webservice.impl.ComprehensiveCreditWS" />
 	<bean id="weiXinWS" class="com.creditease.webservice.impl.WeiXinWS" />
+	<bean id="complianceCheckImpl" class="com.creditease.webservice.impl.ComplianceCheckImpl" />
 	<!-- CXF 配置WebServices的服务名及访问地址 -->
 	<bean id="outLoggingInterceptor" class="org.apache.cxf.interceptor.LoggingOutInterceptor" />
 	<!-- 注意下面的address，这里的address的名称就是访问的WebService的name -->
@@ -54,4 +55,17 @@
 			<ref bean="outLoggingInterceptor" />
 		</jaxws:outInterceptors>
 	</jaxws:server>
+
+	<!--start 客服检查（apply for 综合信贷）-->
+	<jaxws:server id="icomplianceCheckWS"
+				  serviceClass="com.creditease.webservice.IComplianceCheckWS"
+				  address="/complianceCheckWS">
+		<jaxws:serviceBean>
+			<ref bean="complianceCheckImpl"/>
+		</jaxws:serviceBean>
+		<jaxws:outInterceptors>
+			<ref bean="outLoggingInterceptor"/>
+		</jaxws:outInterceptors>
+	</jaxws:server>
+	<!--end 客服检查（apply for 综合信贷）-->
 </beans>
Index: src/main/java/com/creditease/rc/service/impl/ComCheServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/rc/service/impl/ComCheServiceImpl.java	(revision )
+++ src/main/java/com/creditease/rc/service/impl/ComCheServiceImpl.java	(revision )
@@ -0,0 +1,117 @@
+package com.creditease.rc.service.impl;
+
+import com.creditease.rc.domain.CodeTable;
+import com.creditease.rc.service.IComCheService;
+import com.creditease.rc.service.IComplianceCheckService;
+import com.creditease.rc.service.IDataDictionaryService;
+import com.creditease.rc.vo.ComplianceCheckReq;
+import com.creditease.rc.vo.ComplianceCheckRes;
+import com.creditease.rc.vo.ComplianceCheckVo;
+import com.creditease.rc.vo.DataDictionaryVo;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Resource;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by v-weizhang on 2014-12-17.
+ */
+@Service
+public class ComCheServiceImpl implements IComCheService {
+    @Resource
+    private IComplianceCheckService checkService;
+
+    @Resource
+    private IDataDictionaryService dataDictionaryService;
+
+    @Override
+    public ComplianceCheckRes searchComCheByType(ComplianceCheckReq req) {
+        ComplianceCheckRes res = new ComplianceCheckRes();
+        res.setCode("0");
+        DataDictionaryVo vo = new DataDictionaryVo();
+        vo.setSelectType("E");
+        if (org.apache.commons.lang.StringUtils.isEmpty(req.getCheckType())) {
+            vo.setSection("checkpoint_loan");
+            List<CodeTable> tablesCPL = dataDictionaryService.getCodeTableBySection(vo);
+            vo.setSection("errorFile_compliance");
+            List<CodeTable> tablesErrorFile = dataDictionaryService.getCodeTableBySection(vo);
+
+            if (!tablesCPL.isEmpty() && !tablesErrorFile.isEmpty()) {
+                res.setList(this.codeTable2Dictionary(tablesCPL));
+                res.setListErrorFile(this.codeTable2Dictionary(tablesErrorFile));
+                res.setCode("1");
+                res.setMessage("查询成功");
+            } else {
+                res.setMessage("查询客服检查信息时发生错误");
+            }
+        } else {
+            /*apply for commissioner*/
+        }
+        return res;
+    }
+
+    @Override
+    public ComplianceCheckRes saveComCheByType(ComplianceCheckReq req) {
+        ComplianceCheckRes res = new ComplianceCheckRes();
+        res.setCode("0");
+        if (org.apache.commons.lang.StringUtils.isEmpty(req.getCheckType())) {
+            if (null == req.getList() || req.getList().isEmpty()) {
+                res.setMessage("保存失败，未传入有效数据!");
+            } else {
+                if (null != req.getList().get(0).getCreditApplicationId()
+                        && StringUtils.isNotEmpty(req.getList().get(0).getErrorFile())
+                        && StringUtils.isNotEmpty(req.getList().get(0).getCheckPoint())) {
+                    for (ComplianceCheckVo checkVo : req.getList()) {
+                        checkVo.setOperatorId("com_id");
+                        checkVo.setOperatorName("com_name");
+                        checkVo.setOperateDate(new Date());
+                        checkVo.setCheckType("2");
+                        checkVo.setIsNew("Y");
+                    }
+                    ComplianceCheckVo _checkVo = new ComplianceCheckVo();
+                    _checkVo.setCreditApplicationId(req.getList().get(0).getCreditApplicationId());
+                    _checkVo.setCheckType("2");
+                    _checkVo.setIsNew("N");
+                    /*保存检查点*/
+                    checkService.updateComplianceCheckItemFlag(_checkVo);
+                    checkService.saveComplianceCheckItem(req.getList());
+
+                    _checkVo.setComplianceCheckPointType("0");
+                    Double score = checkService.selectComplianceCheckScoreByCheckPoint(_checkVo);
+                    _checkVo.setComplianceCheckScoreType("1");
+                    ComplianceCheckVo checkVo_score = checkService.selectComplianceCheckScoreByIdAndType(_checkVo);
+                    /*保存得分*/
+                    if (null != checkVo_score) {
+                        _checkVo.setComplianceCheckPointScore(score);
+                        checkService.updateComplianceCheckScoreByIdAndType(_checkVo);
+                    } else {
+                        _checkVo.setComplianceCheckPointScore(score);
+                        checkService.saveComplianceCheckScore(_checkVo);
+                    }
+                    res.setCode("1");
+                    res.setMessage("保存成功");
+                } else {
+                    res.setMessage("保存失败，传入数据不完整!");
+                }
+            }
+        } else {
+            /*apply for commissioner*/
+        }
+        return res;
+    }
+
+    private List<DataDictionaryVo> codeTable2Dictionary(List<CodeTable> tables) {
+        List<DataDictionaryVo> voList = new ArrayList<DataDictionaryVo>();
+        DataDictionaryVo vo;
+        for (CodeTable table : tables) {
+            vo = new DataDictionaryVo();
+            vo.setCodeKey(table.getCodeKey());
+            vo.setCodeValue(table.getCodeVlue());
+            voList.add(vo);
+        }
+        return voList;
+    }
+}
Index: src/main/java/com/creditease/rc/service/IComCheService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/rc/service/IComCheService.java	(revision )
+++ src/main/java/com/creditease/rc/service/IComCheService.java	(revision )
@@ -0,0 +1,14 @@
+package com.creditease.rc.service;
+
+import com.creditease.rc.vo.ComplianceCheckReq;
+import com.creditease.rc.vo.ComplianceCheckRes;
+
+/**
+ * Created by v-weizhang on 2014-12-17.
+ */
+public interface IComCheService {
+
+    public ComplianceCheckRes searchComCheByType(ComplianceCheckReq req);
+
+    public ComplianceCheckRes saveComCheByType(ComplianceCheckReq req);
+}
Index: src/main/java/com/creditease/rc/vo/ComplianceCheckReq.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/rc/vo/ComplianceCheckReq.java	(revision )
+++ src/main/java/com/creditease/rc/vo/ComplianceCheckReq.java	(revision )
@@ -0,0 +1,34 @@
+package com.creditease.rc.vo;
+
+import java.util.List;
+
+/**
+ * Created by v-weizhang on 2014-12-16.
+ */
+public class ComplianceCheckReq {
+    /*
+    检查类型：客服检查、合规检查
+     */
+    private String checkType;
+
+    /*
+    检查点
+     */
+    private List<ComplianceCheckVo> list;
+
+    public String getCheckType() {
+        return checkType;
+    }
+
+    public void setCheckType(String checkType) {
+        this.checkType = checkType;
+    }
+
+    public List<ComplianceCheckVo> getList() {
+        return list;
+    }
+
+    public void setList(List<ComplianceCheckVo> list) {
+        this.list = list;
+    }
+}
Index: src/main/java/com/creditease/webservice/impl/ComplianceCheckImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/webservice/impl/ComplianceCheckImpl.java	(revision )
+++ src/main/java/com/creditease/webservice/impl/ComplianceCheckImpl.java	(revision )
@@ -0,0 +1,27 @@
+package com.creditease.webservice.impl;
+
+import com.creditease.rc.service.IComCheService;
+import com.creditease.rc.vo.ComplianceCheckReq;
+import com.creditease.rc.vo.ComplianceCheckRes;
+import com.creditease.webservice.IComplianceCheckWS;
+
+import javax.annotation.Resource;
+
+/**
+ * Created by v-weizhang on 2014-12-16.
+ */
+public class ComplianceCheckImpl implements IComplianceCheckWS {
+    @Resource
+    private IComCheService comCheService;
+
+    @Override
+    public ComplianceCheckRes searchComCheByType(ComplianceCheckReq req) {
+        return comCheService.searchComCheByType(req);
+    }
+
+
+    @Override
+    public ComplianceCheckRes saveComCheByType(ComplianceCheckReq req) {
+        return comCheService.saveComCheByType(req);
+    }
+}
Index: src/main/java/com/creditease/webservice/IComplianceCheckWS.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/webservice/IComplianceCheckWS.java	(revision )
+++ src/main/java/com/creditease/webservice/IComplianceCheckWS.java	(revision )
@@ -0,0 +1,23 @@
+package com.creditease.webservice;
+
+import com.creditease.rc.vo.ComplianceCheckReq;
+import com.creditease.rc.vo.ComplianceCheckRes;
+
+import javax.jws.WebService;
+
+/**
+ * Created by v-weizhang on 2014-12-16.
+ */
+@WebService
+public interface IComplianceCheckWS {
+
+    /*
+    查询合规检查 默认客服检查
+     */
+    public ComplianceCheckRes searchComCheByType(ComplianceCheckReq req);
+
+    /*
+    保存合规检查 默认客服检查
+     */
+    public ComplianceCheckRes saveComCheByType(ComplianceCheckReq req);
+}
Index: src/main/java/com/creditease/rc/vo/ComplianceCheckRes.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/creditease/rc/vo/ComplianceCheckRes.java	(revision )
+++ src/main/java/com/creditease/rc/vo/ComplianceCheckRes.java	(revision )
@@ -0,0 +1,62 @@
+package com.creditease.rc.vo;
+
+import java.util.List;
+
+/**
+ * Created by v-weizhang on 2014-12-16.
+ */
+public class ComplianceCheckRes {
+    /*
+    响应码
+        0 - 查询失败
+        1 - 查询成功
+     */
+    private String code;
+
+    /*
+    响应信息
+     */
+    private String message;
+
+    /*
+    字典信息 检查点
+     */
+    private List<DataDictionaryVo> list;
+
+    /*
+    字典信息 错误文件
+     */
+    private List<DataDictionaryVo> listErrorFile;
+
+    public List<DataDictionaryVo> getList() {
+        return list;
+    }
+
+    public void setList(List<DataDictionaryVo> list) {
+        this.list = list;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public void setMessage(String message) {
+        this.message = message;
+    }
+
+    public List<DataDictionaryVo> getListErrorFile() {
+        return listErrorFile;
+    }
+
+    public void setListErrorFile(List<DataDictionaryVo> listErrorFile) {
+        this.listErrorFile = listErrorFile;
+    }
+}
